python/tutorials/02-fused-softmax.py

```diff
@@ -89,6 +89,101 @@ def softmax_kernel(
     output_ptrs = output_row_start_ptr + col_offsets
     tl.store(output_ptrs, softmax_output, mask=col_offsets < n_cols)
 
+@triton.jit
+def softmax_kernel_big_row(
+    output_ptr, input_ptr, input_row_stride, output_row_stride, n_cols, **meta
+):
+    """ Specialized kernel for big row size inputs.
+
+        issues & todos:
+          - can't use loop
+          - can't use if conditions
+          - could potentially speed up more if we can do exp_sum() using float8
+    """
+    row_idx = tl.program_id(0)
+
+    # The stride represents how much we need to increase the pointer to advance 1 row
+    row_start_ptr = input_ptr + row_idx * input_row_stride
+
+    BLOCK_SIZE = meta['BLOCK_SIZE'] // 2
+
+    tl_max = tl.zeros((1,), tl.float32) * float('-inf')
+    tl_sum = tl.zeros((1,), tl.float32)
+    #for (seg, action) in iter([(0, "load_max"), (1, "load_max_sum"), (0, "load_sum_write"), (1, "load_write")]):
```

Comment:
this is the loop I would have used.

Reply:
ah, I'm not sure that Triton will support structured bindings, but @ptillet will correct me. If you implement this in an old school way I'm guessing that it should just work

Reply:
right, I didn't try

```
for k in values:
   k1, k2 = k.split()
```

For for values, it seems it only supports `range()`, not a tuple or a list or an explicit iterator. Maybe I am missing something? Is there a unit test somewhere for different kind of for loops?